kernel/core/os.adb

```diff
@@ -224,12 +224,12 @@ is
       mbx_msg : os_mbx_msg_t)
    with
       Global => (In_Out => os_task_rw),
-      Pre => (os_ghost_task_mbx_are_well_formed (dest_id) and then
-              os_mbx_get_mbx_count (dest_id) < os_mbx_count_t'Last),
+      Pre => (os_ghost_task_mbx_are_well_formed (dest_id) and
+              not os_mbx_is_full (dest_id)),
       Post => ((os_mbx_get_mbx_count (dest_id) =
-                 os_task_rw (dest_id).mbx.count'Old + 1) and then
+                 os_task_rw (dest_id).mbx.count'Old + 1) and
               (os_task_rw (dest_id).mbx.mbx_array
-                       (os_ghost_get_mbx_tail (dest_id))
+                       (os_ghost_get_mbx_tail (dest_id))  -- medium: postcondition might fail, cannot prove -> head + count is may be too hard to prove -> proposal: add a tail index in os_mbx_t
```

Comment:
It is a pity to have to add unnecessary things to the program to help the proof. It does not seem like the way to go.

Reply:
I will have a try in a dedicated branch.